src/api/apiClient.ts

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable no-console */
+import axios, { AxiosError } from 'axios'
+
+export const BASE_URL = 'https://api.devcycle.com/'
+// export const BASE_URL = 'http://localhost:4001/'
+export const AUTH_URL = 'https://auth.devcycle.com/'
+
+export const apiClient = axios.create({
+    baseURL: `${BASE_URL}/v1/api`,
+})
+
+export const setDVCReferrer = (
+    command = 'unknown',
+    version: string,
+    caller = 'cli',
+): void => {
+    axios.defaults.headers.common['dvc-referrer'] = 'cli'
+    axios.defaults.headers.common['dvc-referrer-metadata'] = JSON.stringify({
+        command,
+        version,
+        caller,
+    })
+}
+
+apiClient.interceptors.response.use(
+    (response) => {
+        return response
+    },
+    (error: AxiosError) => {
+        if (error.response?.status === 401) {
+            window.dispatchEvent(new Event('sessionExpired'))
+        }
```

Comment:
what is this event used for?

Reply:
not needed, copy paste error, should be error messaged shown to user